x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/advanced_step/advanced_step_form.tsx

```diff
@@ -71,6 +109,22 @@ export const AdvancedStepForm: FC<CreateAnalyticsStepProps> = ({
     randomizeSeed,
   } = form;
 
+  const [numTopClassesOptions, setNumTopClassesOptions] = useState<EuiComboBoxOptionOption[]>([
+    defaultNumTopClassesOption,
+  ]);
+  const [numTopClassesSelectedOptions, setNumTopClassesSelectedOptions] = useState<
+    EuiComboBoxOptionOption[]
+  >(getSelectedNumTomClassesOption(numTopClasses));
+
+  const selectedNumTopClasses =
+    numTopClassesSelectedOptions[0] &&
+    ((numTopClassesSelectedOptions[0].value ?? Number(numTopClassesSelectedOptions[0].label)) as
+      | number
+      | undefined);
+
+  const selectedNumTopClassesIsNaN =
+    selectedNumTopClasses !== undefined && isNaN(selectedNumTopClasses);
```

Comment:
This validation should be enhanced to allow only integer input, and values of -1 or greater. Currently you can enter values like `-25` and `1.234`.

Reply:
Good catch - updated in b6894dda6163f4f0dec5103aaa89066e83cd7840